frundenbot/main.py

```diff
@@ -58,124 +99,110 @@ def __log_incomming_messages(bot, update):
             target_chat += ' (%s)' % chat.username
         LOGGER.info('In:  %s: %s' % (target_chat, update.message.text))
 
-CACHE_REFRESH_TIME = Summary('frunde_cache_refresh_seconds', 'Time spent refreshing cache')
-FRUNDE_OPEN = Gauge('frunde_status', '1 if Frunde is open,-1 on error, 0 otherwise')
-@CACHE_REFRESH_TIME.time()
-def refresh_cache(context: CallbackContext):
-    global cache
-    try:
-        LOGGER.debug('Refresh cache')
-        r = requests.get('https://watchyour.freitagsrunde.org')
-        r.raise_for_status()
-        if 'Wir sind fuer dich da!' in r.text:
-            FRUNDE_OPEN.set(1)
-            cache = emojize(':white_check_mark: Die Freitagsrunde ist offen!', use_aliases=True)
+
+    START_TIME = Summary('frunde_start_seconds', 'Time spent executing /start handler')
+    @START_TIME.time()
+    def _callback_start(self, update: Update, context: CallbackContext):
+        context.bot.sendMessage(chat_id=update.message.chat_id, text='Egal was du sagst, ich sag nur, ob die Freitagsrunde offen hat! Du kannst mich direkt anschreiben oder inline per @FrundenBot in anderen Chats benutzen.')
+
+
+    @command(name='commands', description='List commands supported by this bot.')
+    def _callback_commands(self, update: Update, context: CallbackContext):
+        text = generate_command_list(update, context)
+        context.bot.send_message(chat_id=update.message.chat_id, text=text, parse_mode=ParseMode.MARKDOWN)
+
+
+    OPEN_TIME = Summary('frunde_open_seconds', 'Time spent executing /open handler')
+    @OPEN_TIME.time()
+    @command(name='open', description='Is the Freitagsrunde open right now?')
+    def _callback_is_open(self, update: Update, context: CallbackContext):
+        context.bot.sendMessage(chat_id=update.message.chat_id, text='{}\nÜbrigens kannst du mit /mate nachgucken, ob es noch Getränke gibt.'.format(cache))
+
+
+    WHOAMI_TIME = Summary('frunde_whoami_seconds', 'Time spent executing /whoami handler')
+    @WHOAMI_TIME.time()
+    def _callback_whoami(self, update: Update, context: CallbackContext):
+        context.bot.sendMessage(chat_id=update.message.chat_id, text='You are: {} ({})'.format(update.message.from_user.name, update.message.chat_id))
+        LOGGER.info('This is: {} ({})'.format(update.message.from_user.name, update.message.chat_id))
+
+
+    GET_DRINKS_TIME = Summary('frunde_get_drinks_seconds', 'Time spent executing /mate (/drinks) handler')
```

Comment:
Does `/drinks` really work? In that case you should also specify it in the `@command` decorator's `name` parameter (pass both command names as a list).

Reply:
Done